railties/lib/rails/generators/rails/app/templates/Dockerfile.tt

```diff
@@ -38,10 +38,18 @@ RUN apt-get update -qq && \
 ARG NODE_VERSION=<%= node_version %>
 ARG YARN_VERSION=<%= dockerfile_yarn_version %>
 ENV PATH=/usr/local/node/bin:$PATH
+<% if yarn_through_corepack? -%>
+RUN curl -sL https://github.com/nodenv/node-build/archive/master.tar.gz | tar xz -C /tmp/ && \
+    /tmp/node-build-master/bin/node-build "${NODE_VERSION}" /usr/local/node && \
+    rm -rf /tmp/node-build-master
+RUN corepack enable
+RUN yarn set version $YARN_VERSION
```

Comment:
I'm not familiar with corepack, what does this do or why is it necessary?

We also want to keep as few layers as possible, so combine this into a single `RUN` statement if possible would be preferred.

Reply:
I'm not familiar to.
The latest yarn doc mentions this : 

> The preferred way to manage Yarn is by-project and through [Corepack](https://yarnpkg.com/corepack), a tool shipped by default with Node.js. Modern releases of Yarn aren't meant to be installed globally, or from npm.

and

> Start by enabling [Corepack](https://yarnpkg.com/corepack), if it isn't already; this will add the yarn binary to your PATH:

https://yarnpkg.com/getting-started/install

Enabling corepark is enough to install Yarn with modern versions.

Reply:
I combined these two lines into a single one and squash the commit